runtime/lua/vim/shared.lua

```diff
@@ -365,7 +365,10 @@ function vim.tbl_get(o, ...)
   if #keys == 0 then
     return
   end
-  for _, k in ipairs(keys) do
+  for i, k in ipairs(keys) do
+    if type(o[k]) ~= 'table' and next(keys, i) then
```

Comment:
The logic here is basically to do an early `return nil` if the current unwrapped value is not a table (i.e. we can't go deeper) _and_ there are more keys to iterate.